src/kps/gui/windows/form/dialogs/MailDialog.java

```diff
@@ -59,15 +59,14 @@ protected boolean validateFields() {
         String from = (String) getValue(FieldNames.LocationFrom);
         String to = (String) getValue(FieldNames.LocationTo);
         Priority priority = (Priority) getValue(FieldNames.Priority);
-        Set<CalculatedRoute> routes = simulation.buildCalculatedRoute(from, to, priority);
-        if(routes == null) {
+        calculatedRoute = simulation.buildCalculatedRoute(from, to, priority);
+        if(calculatedRoute == null) {
             JOptionPane.showMessageDialog(
                     this,
                     String.format("Sorry, no route between %s and %s using %s was found. Please change your origin and destination and try again", from, to, priority.toString()),
                     "No viable route", JOptionPane.WARNING_MESSAGE);
             return false;
         }
-        calculatedRoute = new RouteConfirmationDialog(null, routes).getConfirmedRoute();
```

Comment:
I miss read what you said. I've removed it since the calculated route now returns only the best route it can find. It didn't make sense to keep it when it will only ever show one route.